languages/multi-language/src/main/java/de/jplag/multilang/MultiLanguageParser.java

```diff
@@ -29,8 +43,46 @@ public List<Token> parseFiles(Set<File> files, boolean normalize) throws Parsing
         return results;
     }
 
+    private void registerLanguageMaps(List<Language> languages) {
```

Comment:
High complexity, as [marked by sonar](https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=2315&id=jplag_JPlag&open=AZbN4ypZmLwy2Ak0-sjM). Can you refactor this/split it up to improve the structure?